plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/GradleEnclavePlugin.kt

```diff
@@ -30,30 +31,16 @@ import org.gradle.util.VersionNumber
 import java.nio.file.Files
 import java.nio.file.Path
 import java.nio.file.attribute.PosixFilePermission.OWNER_EXECUTE
-import java.util.*
-import java.util.jar.JarFile.MANIFEST_NAME
-import java.util.jar.Manifest
 import java.util.stream.Collectors.toList
 import javax.inject.Inject
 import kotlin.io.path.*
 
 class GradleEnclavePlugin @Inject constructor(private val layout: ProjectLayout) : Plugin<Project> {
     companion object {
-        private val CONCLAVE_SDK_VERSION = getManifestAttribute("Conclave-Release-Version")
-        private val CONCLAVE_GRAALVM_VERSION = getManifestAttribute("Conclave-GraalVM-Version")
-
-        fun getManifestAttribute(name: String): String {
-            // Scan all MANIFEST.MF files in the plugin's classpath and find the given manifest attribute.
-            val values = GradleEnclavePlugin::class.java.classLoader
-                .getResources(MANIFEST_NAME)
-                .asSequence()
-                .mapNotNullTo(TreeSet()) { it.openStream().use(::Manifest).mainAttributes.getValue(name) }
-            return when (values.size) {
-                1 -> values.first()
-                0 -> throw IllegalStateException("Could not find manifest attribute $name")
-                else -> throw IllegalStateException("Found multiple values for manifest attribute $name: $values")
-            }
-        }
+        private val CONCLAVE_SDK_VERSION =
+            getManifestAttribute(this::class.java.classLoader, "Conclave-Release-Version")
+        private val CONCLAVE_GRAALVM_VERSION =
+            getManifestAttribute(this::class.java.classLoader, "Conclave-GraalVM-Version")
```

Comment:
I'd like to save to have another `getManifestAttribute` hence I am passing the classLoader. Is that ok?

Reply:
The classloader doesn't matter in this scan. You can use the `PluginUtils::class.java.classLoader` in the method and it will give the same results.

Reply:
Fine, is this because they are in the same package right? I find a bit weird to add the import for that, will use `PluginUtils` as you suggest anyway